lib/matplotlib/collections.py

```diff
@@ -407,7 +410,7 @@ def contains(self, mouseevent):
                self._picker is not True  # the bool, not just nonzero or 1
             else self._pickradius)
 
-        if self.axes and self.get_offset_position() == "data":
+        if self.axes:
```

Comment:
This will un-stale the view limits more often, is that OK?

Reply:
Properly fixing that would be something like
```
if self.axes and self.get_transform().contains_branch(self.axes.transData)
```
but this currently returns False because VectorTransform doesn't "contain" transData (i.e. `VectorTransform(transData) - transData` can't be simplified out) -- really we'd need another relationship of "depends_on(transData)" -- or we can just declare that that's what contains_branch means.  (This kind of annoying details is another reason I was not so keen on making VectorTransform public, but heh.)

OTOH it's just a minor perf loss (the unstaling would only be done once) and it's not worse than before the lazyfication of viewLims, so I don't think it's too bad either.

Reply:
Good enough.